### PR TITLE
Add optional filename argument for -l

### PR DESCRIPTION
**Pull request by xR3b0rn**

### CODE REVIEW

File: candump.c [hunk 1]
```diff
@@ -314,15 +314,16 @@ int main(int argc, char **argv)
...
 
 	signal(SIGTERM, sigterm);
 	signal(SIGHUP, sigterm);
 	signal(SIGINT, sigterm);
 
 	last_tv.tv_sec = 0;
 	last_tv.tv_usec = 0;
-
-	while ((opt = getopt(argc, argv, "t:HciaSs:lDdxLn:r:he8T:?")) != -1) {
+
```
Comment (marckleinebudde) [inline candump.c:325]: unrelated change

File: candump.c [hunk 2]
```diff
@@ -662,13 +664,23 @@ int main(int argc, char **argv)
...
-			now.tm_min,
-			now.tm_sec);
+		if (!log_filename) {
+			sprintf(fname, "candump-%04d-%02d-%02d_%02d%02d%02d.log",
+				now.tm_year + 1900,
+				now.tm_mon + 1,
+				now.tm_mday,
+				now.tm_hour,
+				now.tm_min,
+				now.tm_sec);
```
Comment (marckleinebudde) [inline candump.c:674]: assign `log_filename = fname;` here....

File: candump.c [hunk 3]
```diff
@@ -662,13 +664,23 @@ int main(int argc, char **argv)
...
+		} else {
+			int len = strlen(fname);
```
Comment (marckleinebudde) [inline candump.c]: ...remove that `else`...and use `log_filename` in `fopen()`instead

File: candump.c [hunk 4]
```diff
@@ -134,7 +134,7 @@ static void print_usage(char *prg)
 	fprintf(stderr, "         -a          (enable additional ASCII output)\n");
 	fprintf(stderr, "         -S          (swap byte order in printed CAN data[] - marked with '%c' )\n", SWAP_DELIMITER);
 	fprintf(stderr, "         -s <level>  (silent mode - %d: off (default) %d: animation %d: silent)\n", SILENT_OFF, SILENT_ANI, SILENT_ON);
-	fprintf(stderr, "         -l          (log CAN-frames into file. Sets '-s %d' by default)\n", SILENT_ON);
+	fprintf(stderr, "         -l <fname>  (log CAN-frames into file. Sets '-s %d' by default)\n", SILENT_ON);
```
Comment (hartkopp) [inline candump.c:137]: What happens to `candump -l vcan0 vcan1` ??
I don't see, that this way to add an 'optional' filename does not work as it breaks the current behaviour.

Comment (xR3b0rn) [reply candump.c:137]: Yeah I see, `candump -l vcan0 vcan1` would change to `candump -l -- vcan0 vcan1`. So breaking compatibility is no option? Another solution would be to introduce a new depending command line parameter, what would be kind of ugly, I guess... Any other ideas?

Comment (hartkopp) [reply candump.c:137]: Adding options is ok, changing options which might break peoples scripts not.
Even if it looks ugly I would suggest a different option, e.g. `-F <fname> (log into <fname>)` or `-f <fname>` ...

Comment (hartkopp) [reply candump.c:137]: > Another solution would be to introduce a new depending command line parameter, what would be kind of ugly, I guess... Any other ideas perhaps?

What about this suggestion? Would it fit for you?
https://github.com/hartkopp/can-utils/commit/74d9feb1044a0ccae63e93ac828b2fdecc012ae6

Comment (xR3b0rn) [reply candump.c:137]: Looks great.

Comment (hartkopp) [reply candump.c:137]: Thanks @xR3b0rn !
The new option is now applied in master branch.
